express-notes-api/index.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+const jsonMethod = express.json();
+app.use(jsonMethod);
+
+const data = require('./data.json');
+const dataArray = [];
+
+app.get('/api/notes', (req, res) => {
+  for (const key in data.notes) {
+    if (!data.notes[key]) {
+      res.status(200).json(dataArray);
+    } else {
+      dataArray.push(data.notes[key]);
+    }
+  }
+  res.status(200).json(dataArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const iD = req.params.id;
+  if (isNaN(parseInt(iD))) {
+    res.status(400).send({ error: 'id must be a positive integer' });
```

Comment:
A `400 Bad Request` should also trigger if the id is:
- A non-positive number
- A non-integer number